nx/public/sl/components.js

```diff
@@ -111,6 +112,51 @@ class SlButton extends LitElement {
   }
 }
 
+class SlDialog extends LitElement {
+  async connectedCallback() {
+    super.connectedCallback();
+    this.shadowRoot.adoptedStyleSheets = [style];
+  }
+
+  static properties = {
+    title: { type: String },
+    message: { type: String },
+    open: { type: Boolean },
+  };
+
+  updated(props) {
+    if (props.has('open')) {
+      if (this.open) {
+        this.shadowRoot.querySelector('dialog').showModal();
+      } else {
+        this.shadowRoot.querySelector('dialog').close();
+      }
+    }
+  }
+
+  onConfirm() {
+    this.dispatchEvent(new CustomEvent('confirm'));
+  }
+
+  onCancel() {
+    this.dispatchEvent(new CustomEvent('cancel'));
+  }
+
+  render() {
+    return html`
+      <dialog class="sl-dialog">
+        <h2>${this.title}</h2>
+        <p>${this.message}</p>
+        <div class="sl-dialog-actions">
+          <sl-button @click=${this.onCancel} class="primary outline">Cancel</sl-button>
+          <sl-button @click=${this.onConfirm}>Confirm</sl-button>
+        </div>
+      </dialog>
```

Comment:
This is a good addition. I would use a slot for the content, though. Similar to what we did for select.

Our contract w/ `sl-THING` is that it has the same attributes as the native control. If we do provide something as an added benefit (label being the best example so far), it should be additive. A developer shouldn't have to work around our API. If I developer tries to use this dialog, we are forcing them into needing a title, etc.

```
The Nx Super Lite UI library provides a near drop in replacement to your existing HTML elements but provides an enhanced user experience. It helps you build consistent and performant UIs for your web applications.
```